qC.cpp

```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <climits>
+
+using namespace std;
+typedef long long ll;
+
+void update_all_for_jump_n_from(int pos, int jump, ll cost, const vector<ll>& vec, vector<vector<ll>>& min_price_vec){
+    /* debugging prints
+    cout << "pos: " << pos << endl;
+    cout << "jump: " << jump << endl;
+    cout << "cost: " << cost << endl;
+    cout << "-------------" << endl;
+    */
+    jump += 1;
```

Comment:
probably would be smarter to construct class with static member that increases with each method call to update its array